### PR TITLE
feat(mssql): detect schema from SQL

### PR DESCRIPTION
**Pull request by krzysztof-kwitt**

### CODE REVIEW

File: ibis/backends/mssql/datatypes.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    'IMAGE': dt.Binary,
+    'VARBINARY': dt.Binary,
+    # Other data types
+    'UNIQUEIDENTIFIER': dt.UUID,
+    'GEOMETRY': dt.Geometry,
+    'GEOGRAPHY': dt.Geography,
+    # This timestamp datatype is also known as "rowversion", and the original name is really unfortunate.
+    # See:
+    # https://learn.microsoft.com/en-us/sql/t-sql/data-types/rowversion-transact-sql?view=sql-server-ver16
+    'TIMESTAMP': dt.Timestamp,
```
Comment (cpcloud) [inline ibis/backends/mssql/datatypes.py:76]: It seems like this should map to `int64` instead, given the description:

> The rowversion data type is just an incrementing number and does not preserve a date or a time.

**Review by cpcloud — CHANGES_REQUESTED** [review]
Small question about `rowversion`/`timestamp`.

**Comment by cpcloud** [comment]
Other than the comment above, this PR looks good to me!

File: ibis/backends/mssql/datatypes.py (hunk 1 above)
Comment (krzysztof-kwitt) [reply ibis/backends/mssql/datatypes.py:76]: Good catch. I looked at this docs again and i found the following snippet:

> A nonnullable rowversion column is semantically equivalent to a binary(8) column. A nullable rowversion column is semantically equivalent to a varbinary(8) column.

So I think, we should have `dt.Binary` here.

Comment (cpcloud) [reply ibis/backends/mssql/datatypes.py]: Sounds good to me!

**Comment by cpcloud** [comment]
@krzysztof-kwitt Can you squash the `fixup!` commits? Then I'll merge!

**Review by cpcloud — APPROVED** [review]

**Comment by mik-laj** [comment]
@cpcloud Done

**Review by cpcloud — APPROVED** [review]

**Pull request merged**